### PR TITLE
chore(deps): Use googleapis from the Bazel Central Registry.

### PR DESCRIPTION
**Pull request by ryanli**
Most of the patches are already present in [the BCR module](https://registry.bazel.build/modules/googleapis).

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14771)

### CODE REVIEW

File: MODULE.bazel [hunk 1]
```diff
@@ -49,16 +50,10 @@ python.toolchain(
     python_version = "3.11",
 )
 
-bazel_dep(name = "googleapis", version = "0.0.0", repo_name = "com_google_googleapis")
-archive_override(
+single_version_override(
     module_name = "googleapis",
-    integrity = "sha256-qfsB7M2N7tXY0blSHgIgq8/yUc0Kskid9Zk8H3OS5nc=",
```
Comment (dbolduc) [inline MODULE.bazel:55]: It looks like we already updated `googleapis` to a more recent version than is in BCR. And generated code based on the newer protos.

**Review by dbolduc — COMMENTED** [review]
This seems like a good cleanup, but I am not sure we want to wait for BCR in order to update the commit SHA of googleapis we are pinned to.

**Comment by ryanli** [comment]
> This seems like a good cleanup, but I am not sure we want to wait for BCR in order to update the commit SHA of googleapis we are pinned to.

Thanks, I guess this is also why some of the tests are failing. Let me close this then.

**Pull request closed without merging**